El Dorado discovery changes (#40)
osisoft/PI-Adapter-DNP3-Docs: content/overview/principles-of-operation.md

### Discovery

> Discovery for the DNP3 adapter may be configured using the [Discovery](#TODO-link-to-Discovery-page) configuration. The discovery process will create a data selection item for each DNP point that is assigned to Class 0, Class 1, Class 2, or Class 3 on the outstation. Discovery is performed via an [integrity scan](#integrity-scans) which may be demanding in terms of bandwidth and outstation resources. Data collected with this scan for points that already have a corresponding data selection item will be sent to the data endpoint(s). However, data that is returned for points that are not already configured as data selection items will only be used to populate the discovery results and not sent to the data endpoint(s). 

Reviewer: ```suggestion
Discovery for the DNP3 adapter may be configured using the [Discovery configuration](#DiscoveryConfiguration). The discovery process creates a data selection item for each DNP point assigned to either Class 0, Class 1, Class 2, or Class 3 on the outstation. Discovery is performed by an [integrity scan](#integrity-scans) which may be demanding in terms of bandwidth and outstation resources. Data collected with this scan for points that have a corresponding data selection item is sent to the data endpoint(s). However, data returned for points that are not configured as data selection items are only used to populate the discovery results and are not sent to the data endpoint(s). 
```

### Integrity scans

The DNP3 adapter can be configured to perform an integrity scan on startup, when the outstation's event buffer overflows, at a defined interval, or any combination of the three. An integrity scan is also used during the [discovery](#discovery) process. During an integrity scan, the adapter polls the outstations for events and then the current value of all points that are assigned to one of the event classes (or Class 0).
The adapter performs an integrity scan by polling `Object Group 60`. If bandwidth or outstation performance is a concern, carefully consider the value of an integrity scan, as the outstation can respond with data for many more points than the adapter is configured to collect data for. The adapter simply discards any data that it receives without a corresponding data selection item. To retrieve the current value of any points not assigned to an event class, the adapter needs to perform a static scan.

> If a startup integrity scan is configured, it will run whenever the adapter is started or whenever a shutdown is indicated by the outstation's IIN bits. It may also be used for discovery. Alternatively, if no startup integrity scan is configured, but an integrity scan at a set interval is defined, then the first integrity scan will still be performed when the adapter is started. 

Reviewer: ```suggestion
If a startup integrity scan is configured, it runs whenever the adapter is started or whenever a shutdown is indicated by the outstation's IIN bits. It may also be used for discovery. Alternatively, if no startup integrity scan is configured, but an integrity scan at a set interval is defined, then the first integrity scan is performed when the adapter is started. 
```
